prow/gerrit/adapter/adapter.go

```diff
@@ -425,18 +426,43 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 		labels      map[string]string
 		annotations map[string]string
 	}
-
 	var jobSpecs []jobSpec
-
-	changedFiles := client.ChangedFilesProvider(&change)
+	baseSHAGetter := func() (string, error) { return baseSHA, nil }
+	var hasRelatedChanges *bool
+	// This headSHAGetter will return the empty string instead of the head SHA in cases where we can be certain that change does not
+	// modify inrepoconfig. This allows multiple changes to share a ProwYAML cache entry so long as they don't touch inrepo config themselves.
+	headSHAGetter := func() (string, error) {
+		changes, err := client.ChangedFilesProvider(&change)()
+		if err != nil {
+			return "", err
+		}
+		if config.ContainsInRepoConfigPath(changes) {
+			return change.CurrentRevision, nil
+		}
+		if hasRelatedChanges == nil {
+			if res, err := c.gc.HasRelatedChanges(instance, change.ChangeID, change.CurrentRevision); err != nil {
+				return "", err
+			} else {
+				hasRelatedChanges = &res
```

Comment:
should this be `*hasRelatedChanges = res`? The pointer reference is a bit hard to understand:
- `hasRelatedChanges = &res` means the pointer var `hasRelatedChanges` is moved to point to something else, and this might be persisted in subsequent calls?
- `*hasRelatedChanges = res` means the value of the pointer var `hasRelatedChanges` is updated, and subsequent calls will for sure use the updated value?

Reply:
The first is correct. In C a pointer to a local stack variable wouldn't be safe to use after the function returns, but it is conveniently permissible in Go.
The latter would be a segfault since `hasRelatedChanges` is `nil` at the time of assignment. I'm using a pointer so I can distinguish when the value has been populated.